src/humanloop/cli/__main__.py

```diff
@@ -0,0 +1,256 @@
+    sync_client = SyncClient(client, base_dir=base_dir, log_level=logging.DEBUG if verbose else logging.WARNING)
+    
+    click.echo(click.style("Pulling files from Humanloop...", fg=INFO_COLOR))
+    click.echo(click.style(f"Path: {path or '(root)'}", fg=INFO_COLOR))
+    click.echo(click.style(f"Environment: {environment or '(default)'}", fg=INFO_COLOR))
+    
+    if verbose: 
+        # Don't use the spinner in verbose mode as the spinner and sync client logging compete
+        successful_files = sync_client.pull(path, environment)
+    else:
+        with progress_context("Pulling files..."):
+            successful_files = sync_client.pull(path, environment)
+    
+    # Get metadata about the operation
+    metadata = sync_client.metadata.get_last_operation()
+    if metadata:
+        # Determine if the operation was successful based on failed_files
+        is_successful = not metadata.get('failed_files') and not metadata.get('error')
+        duration_color = SUCCESS_COLOR if is_successful else ERROR_COLOR
+        click.echo(click.style(f"Pull completed in {metadata['duration_ms']}ms", fg=duration_color))
+        
+        if metadata['successful_files']:
+            click.echo(click.style(f"\nSuccessfully pulled {len(metadata['successful_files'])} files:", fg=SUCCESS_COLOR))
+            
+            if verbose: 
+                for file in metadata['successful_files']:   
+                    click.echo(click.style(f"  ✓ {file}", fg=SUCCESS_COLOR))
+            else:
+                files_to_display = metadata['successful_files'][:MAX_FILES_TO_DISPLAY]
+                for file in files_to_display:
+                    click.echo(click.style(f"  ✓ {file}", fg=SUCCESS_COLOR))
+
+                if len(metadata['successful_files']) > MAX_FILES_TO_DISPLAY:
+                    remaining = len(metadata['successful_files']) - MAX_FILES_TO_DISPLAY
+                    click.echo(click.style(f"  ...and {remaining} more", fg=SUCCESS_COLOR))
+        if metadata['failed_files']:
+            click.echo(click.style(f"\nFailed to pull {len(metadata['failed_files'])} files:", fg=ERROR_COLOR))
+            for file in metadata['failed_files']:
+                click.echo(click.style(f"  ✗ {file}", fg=ERROR_COLOR))
+        if metadata.get('error'):
+            click.echo(click.style(f"\nError: {metadata['error']}", fg=ERROR_COLOR))
+
+def format_timestamp(timestamp: str) -> str:
+    """Format timestamp to a more readable format."""
+    try:
+        dt = datetime.fromisoformat(timestamp.replace('Z', '+00:00'))
+        return dt.strftime('%Y-%m-%d %H:%M:%S')
+    except (ValueError, AttributeError):
+        return timestamp
+
+@cli.command()
+@click.option(
+    "--oneline",
+    is_flag=True,
+    help="Display history in a single line per operation",
+)
+@handle_sync_errors
+@common_options
+def history(api_key: Optional[str], env_file: Optional[str], base_dir: str, base_url: Optional[str], oneline: bool):
```

Comment:
needless:
- introduces state
- not in initial scope
- user can look into .humanloop to check the outcome of CLI command